app/src/main/java/org/fossasia/openevent/general/utils/ProgressDialog.kt

```diff
@@ -0,0 +1,20 @@
+package org.fossasia.openevent.general.utils
+
+import android.app.Dialog
+import android.content.Context
+import android.view.Window
+import org.fossasia.openevent.general.R
+
+class ProgressDialog(context: Context?) {
+    private val dialog = Dialog(context)
+    init {
+        dialog.setCancelable(false)
+        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
+        dialog.setContentView(R.layout.dialog_progress)
+    }
+
+    fun showOrDismiss(show: Boolean) {
+        if (show && !dialog.isShowing) dialog.show()
+        else if (!show && dialog.isShowing) dialog.dismiss()
+    }
+}
```

Comment:
Utils mean in utils.kt file.

Reply:
Create a method for the same in utils, no need to create a new file.

Reply:
That won't work because loading dialog is set as non-cancel able, so I have to cancel each dialog box too. So, if I make it into a function then I won't be able to cancel the dialog created on previous call. Hence, I need to create an object of the progressdialog for the entire fragment. This solves the issue. If I make the dialog box cancel able, then the user can just click outside the dialog box which will make it pointless as it won't stop user intervention on login/signup

Reply:
In utils - 
```java
fun showProgressDialog(context: Context, show: Boolean) {
    private val dialog = Dialog(context)
    dialog.setCancelable(false)
    dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
    dialog.setContentView(R.layout.dialog_progress)
    if (show && !dialog.isShowing) dialog.show() else dialog.dismiss()
}
```

And just call the method - 
`Utils.showProgressDialog(requireContext(), true)`
`Utils.showProgressDialog(requireContext(), false)`

Hope it should work.

Reply:
No, it doesn't. That's what I wrote in beginning. The thing is, when this is called it creates a val dialog. So, dialog.isShowing will always return false, because you are referring to the recently created dialog. Hence, you will never be able to close a dialog which was created on previous call and keep on creating new dialogs.

Reply:
got it!

Reply:
Then you can use it like that - 

In fragment - 
`val dialog = Utils.progressDialog(requireContext())`

In Utils.kt - 

```java
fun progressDialog(context: Context) : Dialog {
    val dialog = Dialog(context)
    dialog.setCancelable(false)
    dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
    dialog.setContentView(R.layout.dialog_progress)
   return dialog
}
```

And visibility can handle inside the fragment.

Reply:
Can I just move the class to utils.kt?

Reply:
Because making a class will be easier when reusing code.

Reply:
Moreover, class will use less memory.

Reply:
IMO, creating a new class only for a progress dialog is not a good idea, when you can just use a common method instead to remove duplication.

Reply:
@iamareebjamal @nikit19 Please make me correct.

Reply:
> Moreover, class will use less memory

1 class will use less memory than no class?

Reply:
What @liveHarshit is suggesting is making a function which initializes the dialog(just like a class constructor.). And handle the logic in the place where the function is called. This is why I'm suggesting a class so that we can handle the logic and initialization in one place. It will be easier then for someone else to use it any where else. Moreover, class will use less memory as in when an object of progressDialog is made, it will use space for one object. Whereas when that initialize method is called it creates an object and then passes it to the object of calling function. Which means for a small amount of time, two objects are in memory. One of calling function and one of initialize function itself. Please correct me if I am wrong.

Reply:
If two classes are invoking two dialogs, the amount of classes are going to be in memory in any case. There is no need to create a class. Make a function which returns a dialog

Reply:
Okay

Reply:
Done